ledger/controllers/grants.js

```diff
@@ -1114,24 +1114,51 @@ async function chooseFromAvailablePromotions (debug, runtime, {
   return underscore.omit(promotion, unnecessaryPromotionDetails)
 }
 
-async function chooseFromAvailablePromotionsByType (debug, runtime, {
-  entries
-}) {
-  const ads = underscore.filter(entries, ({
-    type
-  }) => type === 'ads')
-  const ugp = underscore.filter(entries, ({
-    type
-  }) => {
-    return type === '' || type == null || type === 'ugp'
-  })
-  const adsPromotions = await chooseFromAvailablePromotions(debug, runtime, {
-    entries: ads
-  })
-  const ugpPromotions = await chooseFromAvailablePromotions(debug, runtime, {
-    entries: ugp
-  })
-  return [].concat(adsPromotions, ugpPromotions)
+async function chooseFromAvailablePromotionsByType (debug, runtime, options) {
+  let promotions = []
+  const promoCheckers = [
+    getAdsPromos,
+    getUgpPromos
+  ]
+  for (let available of promoCheckers) {
+    const promos = await available(debug, runtime, options)
+    promotions = promotions.concat(promos)
+  }
+  return promotions
+}
```

Comment:
have you considered writing this function without the loop?

Reply:
yeah, that's how i started it. it just seemed redundant and made more sense (to me) to produce a loop to add more types as needed